base/samples/Sonar.hpp

```diff
@@ -0,0 +1,433 @@
+     *
+     * This is a bit of a bummer. When aggregating scanning sonar data, this
+     * field has not much sense. However, aggregating scanning sonar is
+     * meaningful only if the sonar does not move, so it is often possible to
+     * find a timestamp that does make sense, but YMMV
+     *
+     * Advice: always set to a meaningful value as a lot of processing-related
+     * algorithms expect one value. If you can't have *one*, then split your
+     * scanning sonar into separate Sonar structure (one per beam).
+     */
+    base::Time time;
+
+    /** The time at which the beam(s) acquisition started
+     *
+     * This is useful only if each beam have different acquisition times. If
+     * not, the beam time is taken from the time field.
+     *
+     * @see getBeamAcquisitionStartTime 
+     */
+    std::vector<base::Time> timestamps;
+    
+    /** "Size" of one bin
+     *
+     * Sonars basically "slice" the echo return in the time domain, in what we
+     * call "bins". Each bin has a starting point that is equal to
+     *
+     *     bin_duration * bin_index + beam_acquisition_time
+     *
+     * and is itself bin_duration wide. Conversion to the space domain requires
+     * to know the speed of sound.
+     *
+     * @see getBeamAcquisitionStartTime getBinStartTime getBinStartDistance
+     */
+    base::Time bin_duration;
+    
+    /** Opening of the beam orthogonal to the device's Z direction
+     *
+     * This is usually along the scanning axis (for a scanning sonar), or in the
+     * common plane of all the beams (for a multibeam sonar)
+     */
+    base::Angle beam_width;
+    
+    /** Opening of the beam along the device's Z direction
+     *
+     * This is usually orthogonal to the scanning axis (for a scanning sonar),
+     * or orthogonal to the common plane of all the beams (for a multibeam sonar)
+     */
+    base::Angle beam_height;
+
+    /** Each beam's bearing
+     *
+     * This is the position of the beam center with respect to the "front" of
+     * the sonar (which is at 0)
+     */
+    std::vector<base::Angle> bearings;
+
+    /** The speed of sound in water at the time of acquisition and in m/s*/
+    float speed_of_sound;
+    
+    /** Number of bins in a beam */
+    uint32_t bin_count;
+    
+    /** Number of beams in the structure */
+    uint32_t beam_count;
+
+    /** The bin values
+     *
+     * Bin values must be normalized, i.e. gain effects must be corrected and -
+     * as much as made possible by the sonar's specification - converted to a
+     * linear scale where 1.0 means that the signal received by the transducer
+     * is equal to the signal transmittd.
+     *
+     * The data is stored beam-first, that is the data for beam N starts at N *
+     * bin_count and ends at (N + 1) * bin_count
+     */
+    std::vector<float> bins;
+
+    static float getSpeedOfSoundInWater() { return 1497.0; }
+
+    Sonar() : speed_of_sound(1497), bin_count(0), beam_count(0) {}
```

Comment:
Is there a reason why this constructor doesn't use the default speed of sound value provided by the static method?